packages/gestalt/src/IconButton.js

```diff
@@ -164,35 +183,34 @@ const IconButtonWithForwardRef: React$AbstractComponent<unionProps, unionRefs> =
     setHovered(false);
   };
 
-  if (props.role === 'link') {
-    const { href, rel, target } = props;
-
-    return (
-      <InternalLink
-        accessibilityLabel={accessibilityLabel}
-        disabled={disabled}
-        href={href}
-        onClick={handleLinkClick}
-        onBlur={handleOnBlur}
-        onFocus={handleOnFocus}
-        onMouseDown={handleOnMouseDown}
-        onMouseUp={handleOnMouseUp}
-        onMouseEnter={handleOnMouseEnter}
-        onMouseLeave={handleOnMouseLeave}
-        ref={innerRef}
-        rel={rel}
-        tabIndex={tabIndex}
-        target={target}
-        wrappedComponent="iconButton"
-      >
-        {renderPogComponent()}
-      </InternalLink>
-    );
-  }
-
-  const { accessibilityControls, accessibilityExpanded, accessibilityHaspopup, selected } = props;
+  const linkIconButton = (href, rel, target) => (
```

Comment:
nit: since this is a function, the name should indicate an action, like `buildLinkIconButton` or something